radicle-cli/src/commands/ls.rs

```diff
@@ -0,0 +1,66 @@
+use std::ffi::OsString;
+
+use crate::terminal as term;
+use crate::terminal::args::{Args, Error, Help};
+
+use radicle::prelude::*;
+use radicle::storage::{ReadRepository, WriteStorage};
+
+pub const HELP: Help = Help {
+    name: "ls",
+    description: "List radicle projects and other objects",
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad ls [<option>...]
+
+Options
+
+    --help    Print help
+"#,
+};
+
+pub struct Options {}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+
+        if let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((Options {}, vec![]))
+    }
+}
+
+pub fn run(_options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+    let profile = ctx.profile()?;
+    let storage = &profile.storage;
+    let mut table = term::Table::default();
+
+    storage.projects()?.into_iter().for_each(|id| {
+        let repo = storage.repository(id).unwrap();
+        let (_, head) = repo.head().unwrap();
+        if let Ok(Doc { payload, .. }) = repo.project_of(profile.id()) {
+            let head = format!("{:.7}", head);
```

Comment:
I think we have a `terminal::format::oid`

Reply:
[diff](https://github.com/radicle-dev/heartwood/compare/a6fa21d31d800a23163c107e3588ce9fe28b47fd..728063b6ff6d6896168e623b72117fd5dfbc75c3)